### DISCUSSION TITLE
Module for SSR cache

### DISCUSSION THREAD

**Discussion by bhaskarGyan** [nuxt/nuxt#16414] [General]
I have published an npm package for caching the server-rendered HTML with the option to cache based on **device type** (Web, mobile, crawler) and/or **Headers** . 

Similar to the [nuxt-perfect-cache](https://www.npmjs.com/package/nuxt-perfect-cache) and [nuxt-ssr-cache](https://www.npmjs.com/package/nuxt-ssr-cache) which is available for Nuxt2.

Currently, the caching is **In Memory** using [cache-manager](https://www.npmjs.com/package/cache-manager), With the plan to add **Redis caching** 

Please try the package if it fits your use case and please let me know if you have any feedback/ issue 

**Package details** -> [nuxt-cache-ssr](https://www.npmjs.com/package/nuxt-cache-ssr)

Edit:
**Performance benchmark** of Actual Production app

https://twitter.com/anotherJsGuy/status/1589632390150119431

**Comment by spelling203** [comment]
Excuse me, how to cache the home page

**Reply by bhaskarGyan** [reply]
you can include `'/'` in page option as 
```
 pages: [ '/', '/otherpage ]
```

https://github.com/bhaskarGyan/movies/blob/main/nuxt.config.ts#L48

**Reply by bhaskarGyan** [reply]
added in [v1.0.9](https://www.npmjs.com/package/nuxt-cache-ssr/v/1.0.9)

**Reply by spelling203** [reply]
Thank you. Problem solved.

**Reply by spelling203** [reply]
How do I use disableCacheOnDemand? thanks

**Reply by bhaskarGyan** [reply]
this is experimental,

You can add a header key to the response and it will ignore the cache if the header key is present in res.
https://github.com/bhaskarGyan/nuxt-cache-ssr/blob/main/src/runtime/cache.utils.ts#L13


use-case can be, 

say you are making 3 API call to create the page and you want to cache the page only if all the API call is successful.
on failure of any of the API call on server, you can [appendHeader](https://github.com/nuxt/framework/blob/main/packages/nuxt/src/app/composables/cookie.ts#L76)  to your response, and the page will be ignored from the cache.

or based on some of your logic in your own server middleware you can add a header in res (Same as what is configured in nuxt.config) to disable cache on the particular request.
